Modlog

olorin99 opened the pull request:
Adds modlog for Mbin and lemmy (Piefed doesn't have API support yet).
Modlog is available from community menu for Mbin and from both community and user menus for Lemmy.
Also have a instance level modlog under the 'About' page in settings. We should probably add an instance page with instance info (admins, rules, etc) where this would be better placed but for now I figured under About made most sense.

Still needs a lot of testing but some feedback would be good. Should I include the deleted post/comment in the UI since you sort of expect to see unpleasant stuff in the modlog or should I obscure them.

<img width="630" height="1318" alt="Screenshot from 2025-12-22 23-00-46" src="https://github.com/user-attachments/assets/3c825214-4791-42fa-8fa4-7a631bd16ed9" />

jwr1 commented:
Sure, I'm a bit busy this week, but I'll take a look when I get a chance! At first glance, since the mod log is quite a dense view in general, maybe we'd want to ditch the cards and just use dividers between items (or at least see what it looks like).

olorin99 commented:
Removed the cards and added some dividers. I reckon this does look better.
<img width="633" height="1336" alt="Screenshot from 2025-12-27 23-54-15" src="https://github.com/user-attachments/assets/0ed0824b-8586-4c89-ada9-5fb3f2ba0eb9" />

jwr1 commented:
Yeah, it does seem like we'd need to remove the dividers that are within each report, though. It took me a minute while looking at the screenshot, because it looked like the sections with "Content:" were a whole separate report. Besides that I think it looks pretty good, and I like the color coded flairs that are used for the different mod log types.

olorin99 commented:
Think I've gotten everything. Mbin has less functionality as some things aren't returned in the API response. Mod added/removed and post pinned/unpinned won't link to the corresponding item as the user and post aren't available in the response. I think everything else should work though.

olorin99 commented:
Made an issue [here](https://github.com/MbinOrg/mbin/issues/1931).

The user/community info should wrap now when there's not enough space. 
<img width="353" height="264" alt="Screenshot from 2026-01-03 22-07-41" src="https://github.com/user-attachments/assets/1af72e10-71c9-44c8-9388-ee83cb8939f1" />

It's the same widget thats used for posts and comments so the user/community will wrap there too.
<img width="344" height="471" alt="Screenshot from 2026-01-03 22-06-49" src="https://github.com/user-attachments/assets/a87ef6e2-0aed-4248-8a69-d4172999f16f" />

jwr1 commented:
> Made an issue [here](https://github.com/MbinOrg/mbin/issues/1931).

Thank you!

> The user/community info should wrap now when there's not enough space.

I could be remembering wrong, but I believe previously the content info widget should have truncated the community and user names to prevent overflowing, so I thought we could just do the same here. I could work on that though, unless you want to.

olorin99 commented:
That would be better. I looked through the diff of #304 with the post layout change and couldn't see where the user/community names would be truncated and can't think of any other PRs where it might have changed but if you have an idea of how to do it then that would be great.

olorin99 commented:
Awesome thanks. Was just missing correct placement of `Expanded` and `Flexible`.
